src/main/scripts/src/app/app.component.ts

```diff
@@ -18,14 +18,36 @@
  * -----------------------------------------------------------------------/
  */
 
-import {Component, ViewEncapsulation} from '@angular/core';
+import {Component, OnInit, ViewEncapsulation} from '@angular/core';
+import {Router, RoutesRecognized} from '@angular/router';
+
+import {WorkspaceService} from './core/workspace.service';
 
 @Component({
   selector: 'app',
   templateUrl: './app.component.html',
   styleUrls: ['../styles/basic.scss', './app.component.scss'],
   encapsulation: ViewEncapsulation.None
 })
-export class AppComponent {
+export class AppComponent implements OnInit {
+
+  constructor(private workspaceService: WorkspaceService,
+              private router: Router) {
+  }
+
+  public ngOnInit(): void {
+    this.router.events.subscribe(val => {
```

Comment:
I would call this variable `event` instead of `val` since it is not obvious what it is at first sight.